keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -146,6 +150,16 @@ In its first implementation the risks are low as it tries to be a CRI API
 change with minimal changes to the kubelet and it is gated by the feature
 gate `ContainerCheckpoint`.
 
+One possible risk that was identified during Alpha is that the disk of
```

Comment:
So is the recommendation that we won't promote this to GA without some kind of operator that users can install in conjunction with this feature?

It may be nice to have something in place to avoid this (if not an operator). Some ideas could be a remove all checkpoints in case of disk pressure. Rather than implementing a gc mechanism in kubelet. This is pretty strong but I think we want to avoid configurable gc mechanisms for checkpoint in Kubelet.

Maybe we could add a function to eviction manager in case of disk pressure to remove all checkpoints (if feature is enabled).

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/eviction/helpers.go#L1201

Reply:
So why this is important is that when a node experiences disk pressure in kubelet, then we mark the node as not healthy. GC will trigger and try to make the node healthy again. If kubelet is not taught to either look in the checkpoint directories or remove them, then the node could be kept un healthy. This blocks any users from submitting pods to that node and keeps the node out.

We could document it and say "please delete checkpoints in the location" but we should explict say that in this KEP. An operator is useful but we should at least have some protection on this if the feature is turned on. 

It can be an implementation we add in beta promotion (1.30) but I still think we need something here to help.

Reply:
This has been described now with much more detail and a link to a possible operator has been added.